### PR TITLE
PW-297 add notification endpoints

### PR DESCRIPTION
**Pull request by maassenbas**
Added models, notification service, and include tests

### CODE REVIEW

File: src/main/java/com/adyen/model/marketpay/notification/DeleteNotificationConfigurationRequest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * DeleteNotificationConfigurationRequest
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2017-10-11T12:23:08.541Z")
```
Comment (lancergr) [inline src/main/java/com/adyen/model/marketpay/notification/DeleteNotificationConfigurationRequest.java]: we can remove these lines

File: src/main/java/com/adyen/model/marketpay/notification/InlineResponse200.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+package com.adyen.model.marketpay.notification;
+
+import java.util.Objects;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * InlineResponse200
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2017-10-11T12:23:08.541Z")
+public class InlineResponse200 {
```
Comment (lancergr) [inline src/main/java/com/adyen/model/marketpay/notification/InlineResponse200.java]: is this class needed?

File: src/main/java/com/adyen/model/marketpay/notification/InlineResponse2002.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * InlineResponse2002
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2017-10-11T12:23:08.541Z")
+public class InlineResponse2002 {
```
Comment (lancergr) [inline src/main/java/com/adyen/model/marketpay/notification/InlineResponse2002.java]: I believe we can remove this and also all inline classes such as InlineResponse2002ExchangeMessages

File: src/main/java/com/adyen/service/resource/notification/TestNotificationConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+import java.util.Arrays;
+
+/**
+ * Created by rikt on 9/13/17.
+ */
+public class TestNotificationConfiguration extends Resource {
```
Comment (lancergr) [inline src/main/java/com/adyen/service/resource/notification/TestNotificationConfiguration.java]: let's remove

File: src/main/java/com/adyen/model/marketpay/notification/TestNotificationConfigurationResponse.java [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+    private List<String> errorMessages = null;
+
+    @SerializedName("submittedAsync")
+    private Boolean submittedAsync = null;
+
+    @SerializedName("okMessages")
+    private List<String> okMessages = null;
+
+    @SerializedName("exchangeMessages")
+    private List<InlineResponse2002ExchangeMessages> exchangeMessages = null;
```
Comment (lancergr) [inline src/main/java/com/adyen/model/marketpay/notification/TestNotificationConfigurationResponse.java]: shouldn't we use ExchangeMessage instead?

File: src/main/java/com/adyen/service/resource/notification/UpdateNotificationConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package com.adyen.service.resource.notification;
+
+import com.adyen.Service;
+import com.adyen.service.Resource;
+
+import java.util.Arrays;
+
+/**
+ * Created by rikt on 9/13/17.
```
Comment (lancergr) [inline src/main/java/com/adyen/service/resource/notification/UpdateNotificationConfiguration.java]: let's remove these comments as well or replace with class description javadoc

File: src/test/java/com/adyen/MarketPayNotificationTest.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.adyen.model.marketpay.notification.*;
+import com.adyen.service.Notification;
+import org.junit.Test;
+
+public class MarketPayNotificationTest extends BaseTest {
+
+    @Test
+    public void TestCreateNotificationConfiguration() throws Exception {
+        Client client = createMockClientFromFile("mocks/marketpay/notification/createNotificationConfiguration.json");
```
Comment (Aleffio) [inline src/test/java/com/adyen/MarketPayNotificationTest.java]: Should we use "create-notification-configuration-success.json" for the json files like in the other mocks?

File: src/main/java/com/adyen/service/resource/notification/TestNotificationConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Arrays;
+
+
+public class TestNotificationConfiguration extends Resource {
+
+    public TestNotificationConfiguration(Service service) {
+        super(service,
+              service.getClient().getConfig().getMarketPayEndpoint() + "/Notification/" + service.getClient().getMarketPayApiVersion() + "/testNotificationConfiguration",
+              Arrays.asList(
+                      "configurationDetails"
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/resource/notification/TestNotificationConfiguration.java]: Should this be "notificationId", "eventTypes"?

File: src/main/java/com/adyen/service/resource/notification/GetNotificationConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.Arrays;
+
+public class GetNotificationConfiguration extends Resource {
+
+    public GetNotificationConfiguration(Service service) {
+        super(service,
+              service.getClient().getConfig().getMarketPayEndpoint() + "/Notification/" + service.getClient().getMarketPayApiVersion() + "/getNotificationConfiguration",
+              Arrays.asList(
+                      "configurationDetails"
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/resource/notification/GetNotificationConfiguration.java]: Should this be "notificationId"?

File: src/main/java/com/adyen/service/resource/notification/DeleteNotificationConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+import com.adyen.service.Resource;
+
+import java.util.Arrays;
+
+
+public class DeleteNotificationConfiguration extends Resource {
+
+    public DeleteNotificationConfiguration(Service service) {
+        super(service,
+              service.getClient().getConfig().getMarketPayEndpoint() + "/Notification/" + service.getClient().getMarketPayApiVersion() + "/deleteNotificationConfiguration",
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/resource/notification/DeleteNotificationConfiguration.java]: The endpoint for this call is /deleteNotificationConfigurations, please add the final 's'

File: src/main/java/com/adyen/service/resource/notification/DeleteNotificationConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+package com.adyen.service.resource.notification;
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/resource/notification/DeleteNotificationConfiguration.java]: Please add the Copyright file on all the /resource/notification files

File: src/main/java/com/adyen/model/marketpay/notification/GenericResponse.java [hunk 12]
```diff
@@ -0,0 +1,113 @@
+/*
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/marketpay/notification/GenericResponse.java]: We already have a GenericResponse.java in: 
src/main/java/com/adyen/model/GenericResponse.java

Is this one different/necessary?

**Pull request merged**